python/main_loop.py

```diff
@@ -112,7 +114,9 @@ def createNewLocalPath(sailbot, maxAllowableRuntimeSeconds, desiredHeadingPublis
             state, numLookAheadWaypoints=utils.NUM_LOOK_AHEAD_WAYPOINTS_FOR_UPWIND_DOWNWIND, showWarnings=True)
         hasObstacleOnPath = localPath.obstacleOnPath(
             state, numLookAheadWaypoints=utils.NUM_LOOK_AHEAD_WAYPOINTS_FOR_OBSTACLES, showWarnings=True)
-        isGlobalWaypointReached = localPath.lastWaypointReached(state.position)
+        nextGlobalWaypoint = sailbot.globalPath[sailbot.globalPathIndex]
+        previousGlobalWaypoint = sailbot.globalPath[sailbot.globalPathIndex - 1]
+        isGlobalWaypointReached = localPath.waypointReached(state.position, previousGlobalWaypoint, nextGlobalWaypoint)
```

Comment:
Would also be great to have a different hardcoded value for global and local waypoint reached, so we can more easily test and debug. Maybe add a default parameter of distance or a default parameter bool of true or false?